model-datastructure/src/commonMain/kotlin/org/modelix/model/operations/OTBranch.kt

```diff
@@ -78,7 +95,21 @@ class OTBranch(
 
     override fun <T> computeWrite(computable: () -> T): T {
         checkNotEDT()
-        return branch.computeWrite(computable)
+        return if (canWrite()) {
+            branch.computeWrite(computable)
+        } else {
+            branch.computeWriteT { t ->
+                try {
+                    val result = computable()
+                    runSynchronized(completedChanges) {
+                        completedChanges += currentOperations to t.tree
+                    }
+                    result
+                } finally {
+                    currentOperations = ArrayList()
+                }
+            }
+        }
```

Comment:
Can you add some explanations on the two conditions here? They are hard to understand without prior knowledge.